packages/event-cache/src/EventCache.js

```diff
@@ -18,18 +18,28 @@ const limiter = new Bottleneck({ maxConcurrent: 25 })
 
 const getPastEvents = memoize(
   async function(instance, fromBlock, toBlock, batchSize = 10000) {
-    if (!instance.loadedCache && instance.ipfsEventCache) {
+    if (
+      instance.ipfsEventCache &&
+      !instance.loadedCache &&
+      (!instance.cacheMaxBlock ||
+        instance.latestIndexedBlock < instance.cacheMaxBlock)
+    ) {
       try {
         debug('Loading event cache from IPFS', instance.ipfsEventCache)
         const cachedEvents = flattenDeep(
           await Promise.all(
             instance.ipfsEventCache.map(hash => get(instance.ipfsServer, hash))
           )
         )
-        fromBlock = cachedEvents[cachedEvents.length - 1].blockNumber + 1
-        debug(`Last cached blockNumber: ${fromBlock}`)
+        const lastCached = cachedEvents[cachedEvents.length - 1].blockNumber + 1
```

Comment:
Nit: to make it a bit easier to reason about, I would not add +1 here and then remove -1 on line #38 and add +1 on line 41